### ISSUE TITLE
How to load a js from theme before core js scripts?

### ISSUE THREAD

**Issue by evalenzuela** [yiisoft/yii2#11815]
Due project requirements I have to use a different version of jQuery than the one which comes with Yii2. I know I can disable it by using:    

    'assetManager' => [
            'bundles' => [
                // you can override AssetBundle configs here       
                 'yii\web\JqueryAsset' => [
                    'sourcePath' => null,
                    'js' => [] 
                ],        
            ],
        ],

Then I go to AppAsset:

    class AppAsset extends AssetBundle
    {
        public $sourcePath = "@app/themes/v1/assets";
    
        public $jsOptions = [ 'position' => \yii\web\View::POS_END ];
    
        public $js = [
            "js/lib/jquery/jquery-1.9.1.js",
           
        ]; 
    }

But this brings one big issue, jquery loads after yii.validation, etc:

    </script><script src="/assets/5b466ff1/yii.js?v=1465857632"></script>
    <script src="/assets/5b466ff1/yii.validation.js?v=1465857632"></script>
    <script src="/assets/5b466ff1/yii.activeForm.js?v=1465857632"></script>
    <script src="/assets/98d185b3/js/lib/jquery/jquery-1.9.1.js?v=1466702133"></script>

My temporal solution is changing POS_END to POS_HEAD, but that affects page SEO right?, as far as I know is considered bad practice to load js on head.

So my question is:

How can I load jQuery from my theme assets, using POS_END and loading before yii.validation etc?

A possibility is to add it on web/js/jquery.js but I would prefer to keep it on the assets bundle.

Any ideas?

**Comment by agungsijawir** [comment]
you disable JqueryAsset but didn't specify which jquery file you are using.
Look here:
http://www.yiiframework.com/doc-2.0/guide-structure-assets.html#customizing-asset-bundles

```php
'assetManager' => [
            'bundles' => [
                'yii\web\JqueryAsset' => [
                    'sourcePath' => null,   // do not publish the bundle
                    'js' => [
                        '//ajax.googleapis.com/ajax/libs/jquery/2.1.1/jquery.min.js', // here specify your version of jquery
                    ]
                ],
            ],
        ],
```

By doing this, you can safely depends your assets to \yii\web\JqueryAsset with your jquery version.

Please close this issue as this issue is question. Use forum / irc to ask such a question.

**Comment by evalenzuela** [comment]
Thanks for your answer. I knew that, but how I set it to use the jquery file inside my theme? so I would be able to use the jquery url generated by the bundle, something like: /assets/98d185b3/js/jquery.js. In that way I can take advantage of compression, etc.

**Comment by evalenzuela** [comment]
In case someone else needs the answer: http://stackoverflow.com/questions/38005032/yii-2-and-jquery/38005950#38005950
